worldedit-core/src/main/java/com/sk89q/worldedit/command/UtilityCommands.java

```diff
@@ -510,30 +511,36 @@ public void remove(Actor actor, CommandContext args) throws WorldEditException,
     }
 
     @Command(
-        aliases = { "/calc", "/calculate", "/eval", "/evaluate", "/solve" },
-        usage = "<expression>",
-        desc = "Evaluate a mathematical expression"
+            aliases = { "/calc", "/calculate", "/eval", "/evaluate", "/solve" },
+            usage = "<expression>",
+            desc = "Evaluate a mathematical expression"
     )
     @CommandPermissions("worldedit.calc")
-    public void calc(Actor actor, @Text String input) throws CommandException {
-        try {
-            Expression expression = Expression.compile(input);
-            actor.print("= " + expression.evaluate());
-        } catch (EvaluationException e) {
-            actor.printError(String.format(
-                    "'%s' could not be parsed as a valid expression", input));
-        } catch (ExpressionException e) {
-            actor.printError(String.format(
-                    "'%s' could not be evaluated (error: %s)", input, e.getMessage()));
-        }
+    public void calc(final Actor actor, @Text final String input) throws CommandException {
+        // executing as a thread should keep the server from crashing when using for loops.
+        Executors.newCachedThreadPool().execute(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Expression expression = Expression.compile(input);
+                    actor.print(input + " = " + expression.evaluate());
+                } catch (EvaluationException e) {
+                    actor.printError(String.format("'%s' could not be parsed as a valid expression", input));
+                } catch (ExpressionException e) {
+                    actor.printError(String.format("'%s' could not be evaluated (error: %s)", input, e.getMessage()));
+                }
+                // shuts the thread down so its not sitting somewhere aimlessly after executing fully
+                Thread.currentThread().stop();
```

Comment:
Thread.stop is completely deprecated and irresponsible to use when using an executor. Rely on the executor service to shut down the thread using ExecutorService.shutdown.

Reply:
ok, will change the code and i am not sure how the formatting is happening.